downstairs/src/lib.rs

```diff
@@ -1074,9 +1074,11 @@ async fn proc_task(
 
             new_read = msg_channel_rx.recv() => {
                 let mut ds = ads.lock().await;
-                if !ds.on_message_for(id, new_read).await? {
-                    info!(log, "on_message_for returned False; returning now");
-                    return Ok(());
+                let empty = new_read.is_none();
+                ds.on_message_for(id, new_read).await;
```

Comment:
I find this a little confusing as we are deciding if we want to quit based
on the message being `None`, but we still send the message to `on_message_for()`
anyway, which also behaves differently if the message is `None`.  It kind of splits
the logic into two places.  Is this all going to be refactored more later?